asf_public_discourse_home_decarbonisation/analysis/FAQ_analysis/questions_eda_analysis.py

```diff
@@ -0,0 +1,155 @@
+        pd.DataFrame: The loaded data.
+    """
+    return pd.read_csv(file_path)
+
+
+def get_question_counts(df: pd.DataFrame, column_name: str = "Question") -> pd.Series:
+    """
+    Calculates the frequency of each unique question in a specified column.
+
+    Args:
+        df (pd.DataFrame): The DataFrame containing the questions.
+        column_name (str): The name of the column from which to count questions.
+
+    Returns:
+        pd.Series: A Series containing question counts, indexed by the question text.
+    """
+    return df[column_name].value_counts()
+
+
+def wrap_text_with_ellipsis(text: str, line_width: int, max_lines: int) -> str:
+    """
+    Wraps text to a specified line width and maximum number of lines, adding an ellipsis if truncated.
+
+    Args:
+        text (str): The text to wrap.
+        line_width (int): The maximum width of each line.
+        max_lines (int): The maximum number of lines.
+
+    Returns:
+        str: The wrapped (and possibly truncated) text with an ellipsis if necessary.
+    """
+    wrapped_text = textwrap.wrap(text, width=line_width)
+    if len(wrapped_text) > max_lines:
+        return "\n".join(wrapped_text[:max_lines]) + "..."
+    else:
+        return "\n".join(wrapped_text)
+
+
+def plot_question_counts(
+    question_counts: pd.Series, top_n: int = 5, figure_path: str = None
+) -> None:
+    """
+    Plots a horizontal bar chart of the most frequent questions.
+
+    Args:
+        question_counts (pd.Series): A Series containing question counts, indexed by question text.
+        top_n (int): The number of top questions to display.
+    """
+    top_questions = question_counts.head(top_n)
+    wrapped_questions = [wrap_text_with_ellipsis(q, 40, 2) for q in top_questions.index]
+
+    plt.figure(figsize=(12, 8))
+    plt.barh(wrapped_questions, top_questions.values, color=NESTA_COLOURS[0])
+    plt.gcf().subplots_adjust(left=0.5)
+    plt.xlabel("Frequency")
+    plt.ylabel("Questions")
+    plt.title(f"Top {top_n} Most Frequent Questions")
+    plt.gca().invert_yaxis()
+    plt.gca().xaxis.set_major_locator(MaxNLocator(integer=True))
+    plt.savefig(
+        figure_path + "most_frequent_questions.png", dpi=300, bbox_inches="tight"
+    )
+    plt.show()
+
+
+def main():
+    """
+    Main function to perform data loading, question counting, and plotting the most frequent questions.
+    """
+    args = create_argparser()
+    category = args.category
+    forum = args.forum
+    input_data = os.path.join(
+        PROJECT_DIR,
+        f"outputs/data/extracted_questions/{forum}/forum_{category}/extracted_questions_{category}_all.csv",
+    )
+    output_figures_path = os.path.join(
+        PROJECT_DIR, f"outputs/figures/extracted_questions/{forum}/forum_{category}/"
+    )
+    extracted_questions_df = load_data(input_data)
+    question_counts = get_question_counts(extracted_questions_df)
+    plot_question_counts(question_counts, figure_path=output_figures_path)
+
```

Comment:
a few additional things I did/checked (the code is not necessarily the best, just initial thoughts from me!):
```suggestion
    os.makedirs(output_figures_path, exist_ok=True)
    extracted_questions_df = load_data(input_data)
    
    extracted_questions_df["contains_qm"] =extracted_questions_df["Question"].apply(lambda x: True if "?" in x else False)
    print("Contains question mark?\n", extracted_questions_df["contains_qm"].value_counts())
    
    extracted_questions_df["Question"] = extracted_questions_df["Question"].str.lower()
    
    question_counts = get_question_counts(extracted_questions_df)
    plot_question_counts(question_counts, figure_path=output_figures_path, top_n=15)

    dont_knows_path = os.path.join(
        PROJECT_DIR,
        f"outputs/data/extracted_questions/{forum}/forum_{category}/idk_phrases_{category}_all.csv",
    )
    dont_knows_df = load_data(dont_knows_path)
    dont_knows_df["sentences_without_inclusion"] = dont_knows_df["sentences_without_inclusion"].str.lower()
    dk_counts = get_question_counts(dont_knows_df, column_name="sentences_without_inclusion")
    dk_counts = dk_counts[dk_counts> 1]
    if len(dk_counts)>0:
        print(dk_counts)
    else:
        print('No frequent "dont know" expressions found')


```